internal/tiger/cmd/db.go

```diff
@@ -348,6 +351,388 @@ Examples:
+		if readOnly {
+			fmt.Fprintf(outputWriter, "  Read-only enforcement: enabled (permanent, role-based)\n")
+		}
+		if statementTimeout > 0 {
+			fmt.Fprintf(outputWriter, "  Statement timeout: %s\n", statementTimeout)
+		}
+		if len(fromRoles) > 0 {
+			fmt.Fprintf(outputWriter, "  Inherits from: %s\n", strings.Join(fromRoles, ", "))
+		}
+		return nil
+	}
+}
+
+func buildDbCreateRoleCmd() *cobra.Command {
+	var roleName string
+	var readOnly bool
+	var fromRoles string
+	var statementTimeout time.Duration
+	var passwordFlag string
+	var output string
+
+	cmd := &cobra.Command{
+		Use:   "role [service-id]",
+		Short: "Create a new database role",
+		Long: `Create a new database role with optional read-only enforcement.
+
+The service ID can be provided as an argument or will use the default service
+from your configuration.
+
+By default, a secure random password is auto-generated for the new role. You can:
+- Provide an explicit password with --password=<value>
+- Use TIGER_NEW_PASSWORD environment variable
+- Let it auto-generate (default)
+
+The password is saved according to your --password-storage setting (keyring, pgpass, or none).
+
+Read-Only Mode for AI Agents:
+The --read-only flag enables permanent read-only enforcement at the PostgreSQL level
+using the tsdb_admin.read_only_role extension setting. This is designed to provide
+safe database access for AI agents and automated tools that need to read production
+data without risk of modification.
+
+Examples:
+  # Create a role with global database access (uses default service, auto-generates password)
+  tiger db create role --name ai_analyst --from tsdbadmin
+
+  # Create a role for specific service
+  tiger db create role svc-12345 --name ai_analyst
+
+  # Create a read-only role
+  tiger db create role --name ai_analyst --read-only
+
+  # Create a read-only role with same grants as another role
+  tiger db create role --name ai_analyst --read-only --from app_role
+
+  # Create a read-only role inheriting from multiple roles
+  tiger db create role --name ai_analyst --read-only --from app_role,readonly_role
+
+  # Create a read-only role with statement timeout
+  tiger db create role --name ai_analyst --read-only --statement-timeout 30s
+
+  # Create a role with specific password
+  tiger db create role --name ai_analyst --password=my-secure-password
+
+  # Create a role with password from environment variable
+  export TIGER_NEW_PASSWORD=my-secure-password
+  tiger db create role --name ai_analyst
+
+Technical Details:
+This command executes PostgreSQL statements in a transaction to create and configure the role.
+
+CREATE ROLE Options Used:
+  - LOGIN: Always enabled to allow the role to connect
+  - PASSWORD: Always set (from flag, env var, or auto-generated)
+  - IN ROLE: Added when --from flag is provided to inherit grants from existing roles
+
+GUCs (PostgreSQL Configuration Parameters) That May Be Set:
```

Comment:
Not sure that the average user knows what a GUC is, or that mentioning it here provides much value. Why not just:


```suggestion
PostgreSQL Configuration Parameters That May Be Set:
```